app/commands/todo/abandon.rb

```diff
@@ -7,12 +7,17 @@ module Abandon
         class Command
           attr_reader :payload, :aggregate_id
 
+          def self.build(**args)
+            new(**args).tap(&:validate)
+          end
+
           def initialize(params)
             @payload = params.slice(:todo_id, :abandoned_on)
             @aggregate_id = payload.delete(:todo_id)
           end
 
-          def valid?
+          def validate
+            raise BadRequest, 'todo_id is blank' if aggregate_id.nil?
```

Comment:
I don't think the Sinatra error handler deals with ancestors so we'd have to make that more complicated. I'd vote for keeping this simple for the example app.